### PR TITLE
Add group finder preferences and sub-preferences

### PR DESCRIPTION
**Pull request by jordanwade**
## DESCRIPTION
Adds preferences and sub-preferences to be used by community/group finder pages.

### 1. What does this PR do, or why is it needed?

We need to get definedValues for preferences and sub-preferences to feed community/group finder.

### 2. How do I test this PR?

Fire up the web-app-v2 and poke around to `/community` and or `/community/crew` to see data coming in for preferences and sub-preferences

## SCREENSHOTS/GIFS

image
image

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] No new warnings in tests, storybook, or apps
- [x] Assign relevant reviewers
- [x] Get one peer approval before merging

## REVIEW:

**Manual QA**

- [ ] QA branch on Web and ensure it looks/behaves as expected

**Code Review**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: src/data/preferences/schema.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    coverImage: ImageMedia
+  }
+
+  type Preference implements Node {
+    id: ID!
+    title: String
+    summary: String
+    coverImage: ImageMedia
+    url: String
+  }
```
Comment (rkdavidson) [inline src/data/preferences/schema.js:16]: I like the lean-and-mean solution in this PR, so this suggestion is more for us to think about, but...

I wonder if we should put a `subPreferences: [SubPreference]` resolver on `Preference` type?
That way, we could lift the business logic for determining what `SubPreferences` are relevant to each `Preference` up to the API?

I think there's still value in an `allSubPreferences` query, but _could_ see value in that if we run into stumbling blocks on client side soon.

**Review by rkdavidson — APPROVED** [review]
Simple but effective 👌.

I wonder if `GroupPreference` and `GroupSubPreference` would be better names for these things, but there's not a ton of risk on this first pass.

File: src/data/preferences/schema.js (hunk 1 above)
Comment (jordanwade) [reply src/data/preferences/schema.js:16]: Yeah, I think that is a good idea. 👍

**Comment by jordanwade** [comment]
Meant to wait till Monday to merge, but should be fine. Just a new data-source for Preferences and subPreferences.
